### ISSUE TITLE
create a grsecurity policy

### ISSUE THREAD

**Issue by adrelanos** [Whonix/Gentoo-Port#14]
As per initial requirement by @an0ncs0...

(Related, see also: https://forums.grsecurity.net/viewtopic.php?f=5&t=4097)

If I understood @martincmelik right at our recent voice chat, we concluded it would be "insane" time / maintenance requirement wise to create a grsecurity policy for an operating system featuring a desktop environment. And that it would be best after having an initial implementation of Whonix Gentoo Hardened, if @an0ncs0 would try to hire someone from the grsecurity team or otherwise skilled person dedicated for this task.

**Comment by martinholovsky** [comment]
Response from Grsec team sent via email

**Comment by adrelanos** [comment]
Can you please summarize it here, @martincmelik?

**Comment by memorylost731** [comment]
I agree that having support from Grsecurity team is a must. I would happily cover the cost

**Comment by memorylost731** [comment]
in regards to a desktop (which is necessary for wider adoption), we could give a try at balckbox

**Comment by adrelanos** [comment]
Can you please summarize that e-mail here, @martincmelik?

**Comment by naturalspringwater** [comment]
It's worth mentioning, that despite what spender recommends (against) (for good reason, although it's quite forced on his users), containing specific applications with grsec RBAC is in fact possible. It requires exploiting the inheritance in RBAC to apply unconstrained access to each folder /etc/, /usr/, etc. for all subjects to inherit and to get around RBAC's forced requirements on /.

I used this on a system where AppArmor was more appropriate but not installable, and neither was SELinux (but SELinux was madness anyhow). There was a slight advantage was that RBAC is more fully-featured than AA 2.*, although it doesn't support application integration (which as far as I know is only used by Libvirt).

Ultimately AppArmor is more appropriate though it is less featured, which hopefully will change with AppArmor 3.
